### PR TITLE
Authenticate is invoked on client initialization

### PR DESCRIPTION
**Pull request by sebradloff**
There's little value in a client that is not already authenticated.
Making the authenticate method private and called on the initialization
of a NewClient gives the user a better experience. There's no need to
create a NewClient and then try to use it, only to see that it can not
be authenticated. Providing that feedback at initialization will save
the user that time.

The testing strategy is expanded to create a mock for the authorization
portion. The builder library is leveraged to create accessToken structs
which can be marshaled into the responses that we wish the mock api to
return.

### CODE REVIEW

**Comment by sebradloff** [comment]
There's a decent amount to pick apart here, especially along the test strategy side. I created a `testing.go` file as a helper file because I was reading that it is a preferred pattern for helper functions in test. Still obviously learning best practices here but the auto-authenticate takes place on initialization and I increased the test coverage. Didn't take the time to figure out how to force a form encoding failure, but if you have idea, lmk and I'll add in the test case to give us that coverage.

File: testing.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package gremlin
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/lann/builder"
+	httpmock "gopkg.in/jarcoal/httpmock.v1"
```
[the remark this answers is not in the document]
Comment (sebradloff) [reply testing.go]: Added Go dep. Lmk if that looks right to you. As long as you have the repo cloned into $GOPATH/src it should work.

[the remark this answers is not in the document]
Comment (sonnysideup) [reply testing.go]: LGTM

**Comment by sebradloff** [comment]
@sonnysideup LMK if you want me to merge this

**Pull request merged**